### PR TITLE
Fixes some pylint issues and improves data access in dictionaries.

### PR DESCRIPTION
**Pull request by moisesguimaraes**
Signed-off-by: Moisés Guimarães <guimaraes.moises@gmail.com>

### CODE REVIEW

File: src/paho/mqtt/publish.py [hunk 1]
```diff
@@ -19,53 +19,45 @@
 broker, then disconnect and nothing else is required.
 """
 
+import ssl
```
Comment (PierreF) [inline src/paho/mqtt/publish.py]: Why did you add an import ssl ?

File: src/paho/mqtt/subscribe.py [hunk 2]
```diff
@@ -19,53 +19,53 @@
 you to pass a callback for processing of messages.
 """
 
+import ssl
```
Comment (PierreF) [inline src/paho/mqtt/subscribe.py]: ssl is not used. We should just remove the import.

File: src/paho/mqtt/publish.py (hunk 1 above)
Comment (moisesguimaraes) [inline src/paho/mqtt/publish.py]: You're right, I was working on master branch, so I needed to import it. Now it is not needed anymore.

File: src/paho/mqtt/subscribe.py (hunk 2 above)
Comment (moisesguimaraes) [inline src/paho/mqtt/subscribe.py]: done

Comment (Nzbuu) [inline src/paho/mqtt/subscribe.py]: Have you pushed this change yet?

File: src/paho/mqtt/subscribe.py [hunk 3]
```diff
@@ -124,40 +124,25 @@ def callback(callback, topics, qos=0, userdata=None, hostname="localhost",
...
     client.on_connect = _on_connect
 
     if auth is not None:
-        username = auth['username']
-        try:
-            password = auth['password']
-        except KeyError:
-            password = None
-        client.username_pw_set(username, password)
+        client.username_pw_set(auth['username'],
```
Comment (Nzbuu) [inline src/paho/mqtt/subscribe.py]: What about the following?

``
client.username_pw_set(**auth)
```

File: src/paho/mqtt/subscribe.py (hunk 2 above)
Comment (moisesguimaraes) [inline src/paho/mqtt/subscribe.py]: I removed `import ssl` from `publish.py` and forgot to remove from `subscribe.py`.

Now it is really done!

File: src/paho/mqtt/subscribe.py (hunk 3 above)
Comment (moisesguimaraes) [inline src/paho/mqtt/subscribe.py]: I did the same with `will_set()`.

File: src/paho/mqtt/publish.py [hunk 4]
```diff
@@ -23,49 +23,45 @@
...
-            retain = m['retain']
-        except KeyError:
-            retain = False
-    elif type(m) is tuple:
-        (topic, payload, qos, retain) = m
+
+    message = client._userdata.pop()
+
+    if isinstance(message, dict):
+        client.publish(message['topic'],
```
Comment (moisesguimaraes) [inline src/paho/mqtt/publish.py]: @Nzbuu,

Can I do the same here?

client.publish(**message)

If the dictionary has more keys than the function has parameters, we will get an error like this:

TypeError: publish() got an unexpected keyword argument 'foo'

Comment (PierreF) [inline src/paho/mqtt/publish.py]: This is already true for the username_pw_set or will_set. I would say to keep consistent and do the same.

Comment (moisesguimaraes) [inline src/paho/mqtt/publish.py]: done.

Comment (Nzbuu) [inline src/paho/mqtt/publish.py]: @moisesguimaraes. By expanding the dictionary, all of the key/value pairs are passed to the publish method, not just those that are expected. Hence, it will throw an exception if there are unexpected inputs, which I think is the clearest behaviour.

File: src/paho/mqtt/subscribe.py [hunk 5]
```diff
@@ -21,51 +21,50 @@
...
-def _on_message_simple(c, userdata, message):
+def _on_message_simple(client, userdata, message):
     """Internal callback"""
 
     if userdata['msg_count'] == 0:
         return
 
     # Don't process stale retained messages if 'retained' was false
-    if userdata['retained'] == False and message.retain == True:
+    if userdata['retained'] is False and message.retain is True:
```
Comment (PierreF) [inline src/paho/mqtt/subscribe.py]: Yes, I agree.
